lib/gc/Transforms/IterativeTilingAndFusion.cpp

```diff
@@ -656,59 +663,54 @@ void iterativeTilingAndFusionUntilExhaustion(
     return !unTiledOps.empty();
   };
 
-  auto collectSelfTiledOp = [&f, &selfTiledOp]() -> bool {
-    // Reset
-    selfTiledOp.clear();
-    // Walk through funcOp
-    f->walk([&selfTiledOp](Operation *op) {
-      // Target at certain kind of tiled op, such as matmul/conv implemented
-      // by multiple level of nest loops and candidate slices for better
-      // utilization of parallelism and memory hierarchy.
-      if (succeeded(isSelfTiledOp(op))) {
-        selfTiledOp.insert(op);
-      }
-    });
-    return !selfTiledOp.empty();
-  };
+  // Walk through funcOp
+  f->walk([&tiledOps](Operation *op) {
+    if (succeeded(isTiledOpInLoop(op))) {
+      tiledOps.insert(op);
+    }
+  });
 
   // Iterative tiling and fusion until exhaustion.
   while (collectUnTiledOps()) {
     // If existing tiled op before tiling.
-    if (collectSelfTiledOp()) {
+    if (!tiledOps.empty()) {
       // Sort by topology
-      mlir::topologicalSort(selfTiledOp);
+      mlir::topologicalSort(tiledOps);
       // Record if any fusion happens
       bool changed = false;
       // Iteratively fuse in forward and backward fashion.
-      llvm::for_each(selfTiledOp, [&rewriter, &sliceOptions,
-                                   &changed](Operation *tiledOp) {
-        changed |= succeeded(iterativelyFuseProducerAndConsumerOfTiledOp(
-            rewriter, tiledOp, sliceOptions));
-      });
+      llvm::for_each(
+          tiledOps, [&rewriter, &sliceOptions, &changed](Operation *tiledOp) {
+            changed |= succeeded(iterativelyFuseProducerAndConsumerOfTiledOp(
+                rewriter, tiledOp, sliceOptions));
+          });
+      tiledOps.clear();
```

Comment:
The loop will directly return at line 713 even if there are still some untiled ops left because the `tiledOps` has been cleared here.

Reply:
Here `tiledOps.clear();` is in if-branch and used to mark all tiled ops have been visited. But, Line 713 is inside else-branch when `tiledOps` is  empty.  Then, if no new tiled op generated by default tiling, it indeed needs return.

Reply:
Thanks for the explanation! I didn't notice the return is inside the else block.